show/fgnhg.py

```diff
@@ -47,123 +41,154 @@ def active_hops(nhg):
+                output_list = sorted(output_list)
+            if not output_list:
+                ctx.fail("FG_ROUTE table likely does not contain the required entries")
+            nhg_prefix_report = nhip_prefix_map[output_list[0]].split("|")[1]
+            formatted_output_list = ','.replace(',', '\n').join(output_list)
+            table.append([nhg_prefix_report, formatted_output_list])
+            click.echo(tabulate(table, header, tablefmt="simple"))
 
 @fgnhg.command()
 @click.argument('nhg', required=False)
 def hash_view(nhg):
     config_db = ConfigDBConnector()
     config_db.connect()
-    fg_nhg_prefix_table = {}
-    fg_nhg_alias = {}
-    fg_nhg_prefix_table = config_db.get_table('FG_NHG_PREFIX')
-
-    for key, value in fg_nhg_prefix_table.items():
-        fg_nhg_alias[key] = value['FG_NHG']
-
     state_db = SonicV2Connector(host='127.0.0.1')
     state_db.connect(state_db.STATE_DB, False)  # Make one attempt only STATE_DB
-
     TABLE_NAME_SEPARATOR = '|'
     prefix = 'FG_ROUTE_TABLE' + TABLE_NAME_SEPARATOR
     _hash = '{}{}'.format(prefix, '*')
     table_keys = []
-    table_keys = state_db.keys(state_db.STATE_DB, _hash)
     t_dict = {}
-    header = ["FG_NHG_PREFIX", "Next Hop", "Hash buckets"]
+    header = ["FG NHG Prefix", "Next Hop", "Hash buckets"]
     table = []
     output_dict = {}
     bank_dict = {}
-
+    ctx = click.get_current_context()
+    try:
+        table_keys = sorted(state_db.keys(state_db.STATE_DB, _hash))
+    except Exception as e: 
+        ctx.fail("FG_ROUTE_TABLE does not exist!")
+    if table_keys is None:
+        ctx.fail("FG_ROUTE_TABLE does not exist!")
     if nhg is None:
         for nhg_prefix in table_keys:
             bank_dict = {}
             t_dict = state_db.get_all(state_db.STATE_DB, nhg_prefix)
             vals = sorted(set([val for val in t_dict.values()]))
-
             for nh_ip in vals:
                 bank_ids = sorted([int(k) for k, v in t_dict.items() if v == nh_ip])
-
                 bank_ids = [str(x) for x in bank_ids]
-
                 if nhg_prefix in output_dict:
                     output_dict[nhg_prefix].append(nh_ip.split("@")[0])
                 else:
                     output_dict[nhg_prefix] = [nh_ip.split("@")[0]]
                 bank_dict[nh_ip.split("@")[0]] = bank_ids
-
             bank_dict = OrderedDict(sorted(bank_dict.items()))
             nhg_prefix_report = (nhg_prefix.split("|")[1])
-
             for nhip, val in bank_dict.items():
-                formatted_banks = ','.replace(',', '\n').join(bank_dict[nhip])
-                table.append([nhg_prefix_report, nhip, formatted_banks])
-
-        click.echo(tabulate(table, header, tablefmt="grid"))
-
+                displayed_banks = []
+                bank_output = ""
+                formatted_banks = (bank_dict[nhip])
+                for bankid in formatted_banks:
+                    if (len(str(bankid)) == 1):
```

Comment:
Can we add a comment here to explain this logic, I understand that we are doing this to keep the width and printing of bucket # clean with adjusted spacing, but maybe good to have a small comment on it here.

Reply:
sure, will do